components/sidebar.jsx

```diff
@@ -1,31 +1,105 @@
 var React = require('react');
 var Router = require('react-router');
+var classNames = require('classnames');
 var Link = Router.Link;
 var LinkAnchorSwap = require('./link-anchor-swap.jsx');
 
 var Login = require('./login.jsx');
 
+var Subitem = React.createClass({
+  componentDidMount: function() {
+    this.toggleHighlight();
+  },
+  componentDidUpdate: function() {
+    this.toggleHighlight();
+  },
+  toggleHighlight: function() {
+    var isActive = this.getDOMNode().classList.contains("active");
```

Comment:
This should be either from props, or from state. We should *never* have to consult the DOM node (since as far as React cares, that DOM node is an implementation detail).
```
getIniitialState: function() {
  return {
    active: false
  };
},
highlight: function() { this.setState({ active: true; }); },
unhighlight: function() { this.setState({ active: false; }); },
...
render: function() {
  // render this element while taking `this.state.active` into account
}
```